jdaviz/core/freezable_state.py

```diff
@@ -53,6 +55,27 @@ def _reset_x_limits(self, *event):
             self.x_min = x_min
             self.x_max = x_max
 
+    def _convert_units_y_limits(self, old_unit, new_unit):
+        # override glue's _convert_units_y_limits to account
+        # for equivalencies.  This converts all four corners
+        # of the limits to set new limits that contain those
+        # same corners
+
+        if old_unit != new_unit:
+            if old_unit is None or new_unit is None:
+                self._reset_y_limits()
+                return
+
+            x_corners = np.array([self.x_min, self.x_min, self.x_max, self.x_max])
+            y_corners = np.array([self.y_min, self.y_max, self.y_min, self.y_max])
+            spec = Spectrum1D(spectral_axis=x_corners * u.Unit(self.x_display_unit),
+                              flux=y_corners * u.Unit(old_unit))
```

Comment:
There is a way where we can avoid creating a Spectrum1D object here and just pass the `x_corners` as a Quantity object to flux_conversion and have it generate the necessary equivalency to be able to do the conversion:
https://github.com/spacetelescope/jdaviz/commit/1b80edea2f256e2969c2ca5cae7a32f26bd3718d

Reply:
that feels quite hacky to call that `slice`, but I guess maybe its ok since its just internal.  Definitely like saving the overhead of creating the spectrum1d object _just_ to pass quantity arrays.

Reply:
`slice` has been now been updated to `spectral_axis` and I fixed a bug I introduced in `flux_conversion`, just passing the quantity array is now working!